internal/pkg/cli/deploy/env.go

```diff
@@ -89,7 +89,7 @@ func NewEnvDeployer(in *NewEnvDeployerInput) (*envDeployer, error) {
 		env: in.Env,
 
 		templateFS:       template.New(),
-		s3:               s3.New(envRegionSession),
+		s3:               s3.New(envManagerSession),
```

Comment:
I actually don't get why this issue is happening <img src="https://emojis.slackmojis.com/emojis/images/1643515804/18208/meow_tears.png?1643515804" width="25px" /> if the object is uploaded with bucket ownership: 
https://github.com/aws/copilot-cli/blob/4b45d6682865598b368819023f15185b6d572b93/internal/pkg/aws/s3/s3.go#L217

and the bucket allows all actions from the env accounts:
https://github.com/aws/copilot-cli/blob/4b45d6682865598b368819023f15185b6d572b93/internal/pkg/template/templates/app/cf.yml#L74-L82

Then why is there an access denied error? What am I missing? I'd have guessed that as long as the objects are uploaded with BucketOwnership then it wouldn't matter if they're overriden.